File: Example/OnlyID_Demo_Objc/OnlyID_Demo_Objc/ViewController.m
//
//  ViewController.m
//  OnlyID_Demo_Objc
//
//  Created by Jarvis on 2020/6/21.
//  Copyright © 2020 QFish. All rights reserved.
//

#import "ViewController.h"
#import <OnlyID-Swift.h>

#define ClientId @"73c6cce568d34a25ac426a26a1ca0c1e"
#define SecretId @"36c820ba83bb4944a0744208066e8bbf"

@interface ViewController () <OnlyIDOAuthDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    UIBarButtonItem *loginItem = [[UIBarButtonItem alloc] initWithTitle:@"login" style:UIBarButtonItemStylePlain target:self action:@selector(oauth)];
    self.navigationItem.rightBarButtonItem = loginItem;
}

- (void)oauth {
    OnlyIDOAuthConfig *config = [[OnlyIDOAuthConfig alloc] initWithClientId:ClientId view:nil theme:nil state:nil];
    [OnlyID oauthWithConfig:config fromController:nil delegate:self];
}

- (void)onCompleteWithCode:(NSString *)code state:(NSString *)state {
    NSLog(@"Login successed: %@, %@", code, state);
    
    UILabel *label = [UILabel new];
    label.text = code;
    label.frame = CGRectMake(0, 0, self.view.frame.size.width, 300);
    label.numberOfLines = 0;
    label.textAlignment = NSTextAlignmentCenter;
    [self.view addSubview:label];
}

- (void)onErrorWithError:(NSError *)error {
    NSLog(@"Login failed: %@", error);
}

- (void)onCancel {
    NSLog(@"Login canceled");
}

@end

File: OnlyID_Demo/ViewController.swift
//
//  ViewController.swift
//  OnlyID_Demo
//
//  Created by Alex on 2018/4/20.
//  Copyright © 2018年 OnlyID. All rights reserved.
//

import UIKit
import OnlyID_SDK

class ViewController: UIViewController, AuthDelegate {
    @IBOutlet weak var resultLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    @IBAction func login(_ sender: Any) {
        OnlyID.auth("5adac916904be93f3f621003", delegate: self)
    }
    
    func didReceiveAuthResp(errCode: ErrCode, code: String?, state: String?) {
        switch errCode {
        case .cancel:
            resultLabel.text = "用户取消"
        case .networkErr:
            resultLabel.text = "网络错误"
        default:
            resultLabel.text = "code= " + code! + ", state= " + (state ?? "")
        }
    }
}
